code/player/TTTPlayer.Shop.cs

```diff
@@ -78,13 +98,39 @@ public void RequestPurchase(IItem item)
 
             Credits -= itemData.Price;
 
-            item.OnPurchase(this);
+            Utils.GetObjectByType<IItem>(itemType).OnPurchase(this);
+            BoughtItemsSet.Add(itemData.Name);
 
-            ClientSendQuickshopUpdate(To.Single(this));
+            ClientBoughtItem(To.Single(this), itemData.Name);
+        }
+
+        [Event(TTTEvent.Game.RoundChange)]
+        private static void OnRoundChanged(BaseRound _, BaseRound newRound)
+        {
+            if (newRound is PreRound preRound)
+            {
+                foreach (TTTPlayer player in Utils.GetPlayers())
+                {
+                    player.BoughtItemsSet.Clear();
+                }
+            }
+        }
+
+        [ClientRpc]
+        public static void ClientBoughtItem(string itemName)
+        {
+            (Local.Pawn as TTTPlayer).BoughtItemsSet.Add(itemName);
+
+            UpdateQuickShop();
         }
 
         [ClientRpc]
         public static void ClientSendQuickshopUpdate()
```

Comment:
Just noticed this 
`ClientSendQuickshopUpdate` -> `ClientSendQuickShopUpdate` since `UpdateQuickShop`